pkg/collector/runner/runner.go

```diff
@@ -398,18 +398,28 @@ func addWorkStats(c check.Check, execTime time.Duration, err error, warnings []e
 }
 
 func expCheckStats() interface{} {
-	checkStats.M.RLock()
-	defer checkStats.M.RUnlock()
-
-	return checkStats.Stats
+	return GetCheckStats()
 }
 
 // GetCheckStats returns the check stats map
 func GetCheckStats() map[string]map[check.ID]*check.Stats {
 	checkStats.M.RLock()
 	defer checkStats.M.RUnlock()
 
-	return checkStats.Stats
+	// because the returned maps will be used after the lock is released, and
+	// thus when they might be further modified, we must clone them here.  The
+	// map values (`check.Stats`) are threadsafe and need not be cloned.
+
+	cloned1 := make(map[string]map[check.ID]*check.Stats)
+	for k1, v1 := range checkStats.Stats {
+		cloned2 := make(map[check.ID]*check.Stats)
+		cloned1[k1] = cloned2
```

Comment:
A small nit but you probably want to do this assignment after you populate the `cloned2` structure. While it doesn't make any functional difference, it will make the code more consistent and easier to read.